File: frontend/lib/models/lesson.dart
/// Espelha LessonResponse/LessonQuestionResponse/LessonChoiceResponse em
/// backend/app/routers/schemas.py — POST /topic/{id}/lesson (RF-03).
///
/// `LessonChoice` deliberadamente NÃO tem is_correct/explanation — o
/// backend não manda isso antes do usuário responder (RF-04); ver
/// AnswerChoice em answer.dart pra depois de responder.
class LessonChoice {
  final String id;
  final String text;

  const LessonChoice({required this.id, required this.text});

  factory LessonChoice.fromJson(Map<String, dynamic> json) {
    return LessonChoice(id: json['id'] as String, text: json['text'] as String);
  }
}

class LessonQuestion {
  final String id;
  final String prompt;
  final List<LessonChoice> choices;

  const LessonQuestion({required this.id, required this.prompt, required this.choices});

  factory LessonQuestion.fromJson(Map<String, dynamic> json) {
    return LessonQuestion(
      id: json['id'] as String,
      prompt: json['prompt'] as String,
      choices: (json['choices'] as List)
          .map((c) => LessonChoice.fromJson(c as Map<String, dynamic>))
          .toList(),
    );
  }
}

class Lesson {
  final String sessionId;
  final List<LessonQuestion> questions;

  const Lesson({required this.sessionId, required this.questions});

  factory Lesson.fromJson(Map<String, dynamic> json) {
    return Lesson(
      sessionId: json['session_id'] as String,
      questions: (json['questions'] as List)
          .map((q) => LessonQuestion.fromJson(q as Map<String, dynamic>))
          .toList(),
    );
  }
}

File: backend/app/routers/lesson.py
"""`POST /topic/{id}/lesson` — gera uma lição (RF-03)."""

import uuid
from datetime import datetime, timezone

from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session

from app.models.entities import AppUser
from app.repository.db import get_db
from app.routers.deps import get_current_app_user
from app.routers.schemas import (
    LessonChoiceResponse,
    LessonQuestionResponse,
    LessonResponse,
)
from app.services import lesson_service

router = APIRouter(tags=["lesson"])


@router.post("/topic/{topic_id}/lesson", response_model=LessonResponse)
def start_lesson(
    topic_id: uuid.UUID,
    user: AppUser = Depends(get_current_app_user),
    db: Session = Depends(get_db),
) -> LessonResponse:
    now = datetime.now(timezone.utc)
    try:
        lesson = lesson_service.start_lesson(
            db, user.id, topic_id, today=now.date(), now=now
        )
    except PermissionError as exc:
        raise HTTPException(status.HTTP_403_FORBIDDEN, str(exc)) from exc

    return LessonResponse(
        session_id=lesson.session.id,
        questions=[
            LessonQuestionResponse(
                id=lesson_question.question.id,
                prompt=lesson_question.question.prompt,
                # SEM is_correct/explanation aqui — vazaria a resposta
                # certa antes do usuário responder (RF-04). Só voltam em
                # POST /question/{id}/answer.
                choices=[
                    LessonChoiceResponse(id=choice.id, text=choice.text)
                    for choice in lesson_question.choices
                ],
            )
            for lesson_question in lesson.questions
        ],
    )
